src/ir/context.rs

```diff
@@ -1189,9 +1187,8 @@ impl BindgenContext {
         self.compute_cannot_derive_hash();
         self.compute_cannot_derive_partialord_partialeq_or_eq();
 
-        let ret = cb(self);
-        self.in_codegen = false;
-        ret
+        let ret = cb(&self);
+        (ret, self.into())
```

Comment:
I dont think I've grasped all the idioms of Rust, but still, this seems quite unusual to me. Maybe it would be better to just use `(ret, self.options)`?

Reply:
Ah! This is from when I was trying to do the conversion inside `codegen` rather than here, and privacy wouldn't have allowed that. Good eye!